### PR TITLE
Plm webclient

### PR DESCRIPTION
**Pull request by Deruga**

### CODE REVIEW

File: workflow_creation/utilities_plm_webclient.md [hunk 1]
```diff
@@ -1,41 +1,228 @@
...
+## Roles of PLM in CAxMan workflows
+
+PLM functionality is available in CAxMan – and else – as web-services and through a web-client.
+The web-client user’s guide is in [User guide PLM client](./utilities_plm_webclient.md#user-guide-plm-client). The PLM web-client can act as a service within a CAxMan workflow to hand-over files to subsequent services. The invocation of the web-client as part of a CAxMan workflow is described in [this document](./utilities_plm_webclient.md#integrating-the-plm-web-client-in-a-workflow).
+
+Access to the PLM Server via web-services is twofold:
+1.	[Via GSS](../service_implementation/advanced_plm.md#gss).
+1.	[Via EDMopenSimDM native web-services](../service_implementation/advanced_plm.md#native-services--soap-).
+
+These two types of services give you access to, among others, the following functions:
```
Comment (rojsc) [inline workflow_creation/utilities_plm_webclient.md]: Here, also, I'd like to see "GSS" replaced by "PLM API" or similar. I understand that there is a difference between the native PLM API and the "PLM API for GSS", but I still think calling the latter part GSS only is wrong.

File: workflow_creation/utilities_plm_webclient.md [hunk 2]
```diff
@@ -1,41 +1,228 @@
...
+- Storage of parameter values in folders. (Native)
+- Storage of feedback comments in folders. (Native)
+- Approval of folder versions. (Native)
+
+## User guide PLM client
+
+The PLM web client can be directly accessed by the following URL:
+https://caxman.clesgo.net/jotne/STM_web/
+
+In case of direct access (not via workdlow) sign-in is required. The user name is the one you use for the Portal. If you do not have a specific password for the PLM Server, request one by sending an e-mail to caxman@jotne.com .
```
Comment (rojsc) [inline workflow_creation/utilities_plm_webclient.md]: Typo in 
> (not via workdlow)

File: workflow_creation/utilities_plm_webclient.md [hunk 3]
```diff
@@ -1,41 +1,228 @@
...
+
+### List of models
+
+If the PLM web client is launched from a workflow, no extra sign-in is required. The workflow passes a session token to the PLM web client.
+After havingbeen launched the PLM client presents the list of available repositories and models in the left-hand side panel. An example of available product structures is shown below.
```
Comment (rojsc) [inline workflow_creation/utilities_plm_webclient.md]: Missing space in
> havingbeen

File: workflow_creation/utilities_plm_webclient.md [hunk 4]
```diff
@@ -1,41 +1,228 @@
...
+http://www.caxman.eu/apps/jotne/IframeDisplay.owl#displayHtml_Service
+```
+Use this URI to add the service to your workflow.
+
+The following screen shot shows the minimal connections you need to make to
+the PLM file chooser service block:
+<p align="center">
+  <img src="img_plm/wf_editor.png"
+   alt="Minimal connections made to the PLM file chooser service" />
+</p>
```
Comment (rojsc) [inline workflow_creation/utilities_plm_webclient.md]: I cannot comment on the picture itself since it's not part of this pull request, but the workflow input "sessionToken" is wired to "sessionToken" _and_ "serviceID", which is wrong.

File: workflow_creation/utilities_plm_webclient.md [hunk 5]
```diff
@@ -1,41 +1,228 @@
...
+
+The complete set of parameters is explained in the following table:
+
+| Parameter name | Wiring required? | Description |
+| -------------- | --------- | ----------- |
+| `sessionToken` | yes | The session token used for authentication. Should be connected to the workflow input with the same name |
+| `serviceID` | yes | the same as sessionToken |
```
Comment (rojsc) [inline workflow_creation/utilities_plm_webclient.md]: That's wrong. `serviceID` has to be left _unconnected_, the workflow manager will assign a service ID unique for each service execution to that input.

File: workflow_creation/utilities_plm_webclient.md [hunk 6]
```diff
@@ -1,41 +1,228 @@
...
+| Output | Description |
+| -------------- | ----------- |
+| `selectedFile_base64` | GSS identifier of a selected file |
+| `status_base64` | file selection status |
+
+# File selection using the PLM web client
+Workflows in CAxMan typically communicate with one of the provided cloud
+storages, that is, SWIFT or the PLM Server. File interactions are typically handled
+through GSS (Generic Storage Services), but this requires that the services
+already know where a file shall be stored and which folder it be should
```
Comment (rojsc) [inline workflow_creation/utilities_plm_webclient.md]: Typo in:
> ... folder it be should upload to

File: workflow_creation/utilities_plm_webclient.md (hunk 1 above)
Comment (Deruga) [reply workflow_creation/utilities_plm_webclient.md]: I understand the point.
@JochenHaenisch , any idea on what should we use instead?

File: workflow_creation/utilities_plm_webclient.md (hunk 4 above)
Comment (Deruga) [reply workflow_creation/utilities_plm_webclient.md]: For some reason I cannot have rights to update the "PLM_File_Chooser" workflow

File: workflow_creation/utilities_plm_webclient.md (hunk 2 above)
Comment (Deruga) [reply workflow_creation/utilities_plm_webclient.md]: fixed

File: workflow_creation/utilities_plm_webclient.md (hunk 3 above)
Comment (Deruga) [reply workflow_creation/utilities_plm_webclient.md]: fixed

File: workflow_creation/utilities_plm_webclient.md (hunk 5 above)
Comment (Deruga) [reply workflow_creation/utilities_plm_webclient.md]: fixed

File: workflow_creation/utilities_plm_webclient.md (hunk 1 above)
Comment (JochenHaenisch) [reply workflow_creation/utilities_plm_webclient.md]: What about "PLM access through GSS" as title instead of just "GSS"?

Comment (rojsc) [reply workflow_creation/utilities_plm_webclient.md]: I don't think that's correct either. The text behind the link does not explain anything about the usage of GSS, but the usage of the PLM API.

Comment (rojsc) [reply workflow_creation/utilities_plm_webclient.md]: I don't think that's necessary here. File access through GSS is documented, and from that perspective, there is no difference between PLM and any other storage provider.

Comment (JochenHaenisch) [reply workflow_creation/utilities_plm_webclient.md]: New proposal:
for the first title (GSS): PLM API through GSS
for the second title (native ...): PLM API through native services.

Comment (rojsc) [reply workflow_creation/utilities_plm_webclient.md]: Hmm, could you explain what you mean with "through GSS"?
If it means that obtaining the PLM endpoints is _through GSS_ and the rest is _not_ through GSS, I'm fine with it. ;-)

Comment (JochenHaenisch) [reply workflow_creation/utilities_plm_webclient.md]: OK - I have changed the titles and added some more introductory text to both sections. Alexey needs to commit this.
If you are happy with that, will you merge this into our training repository and inform people of the updates?
Thanks and Happy Easter, Jochen

Comment (rojsc) [reply workflow_creation/utilities_plm_webclient.md]: Sorry for my long silence, I hope you had a good Easter time.

Thanks, Jochen, for your contributions, I'm fine with the current status.

**Comment by rojsc** [comment]
From my side, the request can be merged.
Alexey, would you perform the merge?
